syftlib/src/main/java/org/openmined/syft/domain/SyftDataRepository.kt

```diff
@@ -0,0 +1,7 @@
+package org.openmined.syft.domain
+
+import org.pytorch.IValue
+
+interface SyftDataRepository {
```

Comment:
This part of code will most likely align with @pengyuan dataloader api. We have a basic structure there so may be we can already align both the PR sp they could be merged?